src/http_server.cc

```diff
@@ -3628,15 +3648,34 @@ HTTPAPIServer::BADReplyCallback(evthr_t* thr, void* arg, void* shared)
   delete infer_request;
 }
 
+evhtp_res
+HTTPAPIServer::InferRequestClass::RequestFiniHook(
+    evhtp_request* request, void* arg)
+{
+  HTTPAPIServer::InferRequestClass* infer_request =
+      reinterpret_cast<HTTPAPIServer::InferRequestClass*>(arg);
+  if (infer_request->req_ != request) {
+    LOG_ERROR << "[INTERNAL] mismatched request in fini hook";
```

Comment:
I assume this is not expected to happen, so should this return a EVHTP_RES_BADREQ to promote visibility of failure if it does happen?

Reply:
The error code isn't actually checked or acted upon for the hook. Agree for completeness / readability.